docs/content/en/docs/yaml-crd-ref/metricsprovider.md

```diff
@@ -64,6 +65,7 @@ spec:
   * **secretKeyRef**
     * **name:** -- Name of the token for this data provider
     * **key:** -- Key for this data provider
+    * **optional** -- Boolean value that specifies whether the secret `key` is required
```

Comment:
I suggest using the same documentation text that k8s uses:
```suggestion
    * **optional** -- Specify whether the Secret or its key must be defined
```

source: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#secretkeyselector-v1-core

Reply:
done